apps/rig_outbound_gateway/lib/rig_outbound_gateway/kafka/sup.ex

```diff
@@ -40,10 +40,22 @@ defmodule RigOutboundGateway.Kafka.Sup do
   def init(:ok) do
     conf = config()
 
-    client_conf = [
-      auto_start_producers: true,
-      default_producer_config: []
-    ]
+    if conf.ssl == nil and conf.sasl != nil do
+      Logger.warn("SASL is enabled, but SSL is not - credentials are transmitted as cleartext.")
+    end
+
+    add_sasl_option = fn opts ->
+      if conf.sasl, do: Keyword.put(opts, :sasl, conf.sasl), else: opts
+    end
+
+    client_conf =
+      [
+        endpoints: conf.brokers,
+        auto_start_producers: true,
+        default_producer_config: [],
+        ssl: conf.ssl
+      ]
+      |> add_sasl_option.()
```

Comment:
Do you think we will at some point pass argument here? Or is there any other reason why it's not directly passed as ssl config is?

Reply:
`ssl` is normally a keyword list, but when it's disabled it's set to false. The `sasl` argument works differently: if disabled, it must not be present. I couldn't figure out a nice way to write this inline or with a pipe, so I've opted for the function..

Reply:
makes sense